src/balancereader/src/main/java/anthos/samples/financedemo/balancereader/TransactionRepository.java

```diff
@@ -40,6 +40,12 @@
         + "ORDER BY t.transactionId ASC")
     List<Transaction> findLatest(long latestTransaction);
 
-    @Query("SELECT MAX(transactionId) FROM Transaction")
-    long latestId();
+    /**
+     * Returns the id of the latest transaction, or -1 if no transactions exist.
+     */
+    @Query(value = "SELECT COALESCE("
+        + "(SELECT MAX(transaction_id) FROM transactions), "
+        + "CAST ('-1' AS BIGINT))",
+        nativeQuery = true)
+    Long latestId();
```

Comment:
was there a reason for going from long to Long?

Reply:
Long (instead of long) allowed the query to return NULL.  Not as important anymore since I defaulted the query to return -1 by default.

Reply:
Would we have to add any null checks to other parts of the code?

Reply:
We don't have to now since this query returns a default value.

If you think it makes more sense to do NULL checks in the code, I can do that instead of fixing the query.

Reply:
I guess my question is: if we're using a default value and don't expect nulls, why not keep using long?

Reply:
Using `Long` feels less brittle to me.  It protects against this failure mode from any future changes to the query that may make it return Null again.  'Defense in depth' if you will.

I sense there is resistance to having it `Long`.  I'm assuming that you're concerned about performance efficiency.

Reply:
Definitely not worried about performance. But the LedgerReader class is still expecting long instead of Long, so if you make this change here, you should also make it there. And also add null checks there so it fails in an expected way if it does end up being null. (I'm not really sure what the error would look like if the Long was null right now)

Reply:
Basically, if we're confident that query always returns a value (I didn't read the docs, but it seems like it should), I think it's safe to keep using long. If there's a chance it still returns null, we need to change other code to handle that error case.

Reply:
> But the LedgerReader class is still expecting long instead of Long, so if you make this change here, you should also make it there. 

This actually isn't an issue, see "Java autoboxing".

> And also add null checks there so it fails in an expected way if it does end up being null. (I'm not really sure what the error would look like if the Long was null right now)

That is a fair point, if I'm protecting against Null here in TransactionRepository, then I also still need to do it in LedgerReader.

Okay, you've convinced me to just do the check in the Java code.  I've removed the overly complicated SQL query and added null checks in the Java code.

Reply:
> This actually isn't an issue, see "Java autoboxing".

It looks like Autoboxing thows a NullPointerException if the Long is null though, and allowing nulls was the argument for using Long here in the first place

> Okay, you've convinced me to just do the check in the Java code. I've removed the overly complicated SQL query and added null checks in the Java code.

Ok cool, that sounds like a good solution to me, I'll re-review